src/net_processing.cpp

```diff
@@ -1916,9 +1916,15 @@ void PeerManagerImpl::BlockConnected(
     const std::shared_ptr<const CBlock>& pblock,
     const CBlockIndex* pindex)
 {
-    m_orphanage.EraseForBlock(*pblock);
+    // Update this for all chainstate roles so that we don't mistakenly see peers
+    // helping us do background IBD as having a stale tip.
     m_last_tip_update = GetTime<std::chrono::seconds>();
 
+    if (role == ChainstateRole::BACKGROUND) {
```

Comment:
fbd05a9eed1e40be0c4915e2aef98b3771d7af6c

I see why we don't care about mempool stuff here, but `m_block_stalling_timeout` I'm not so sure. Is this logic completely irrelevant for background sync?

Reply:
Good point! The `m_block_stalling_timeout` was added about a year ago, so when preparing these changes I didn't consider it. It looks like we should keep it up to date for both chainstates. Fixed.